pkg/network/kuryr_bootstrap.go

```diff
@@ -0,0 +1,620 @@
+		}
+	}
+
+	masterSgId, err := findSgId(client, "master", tag)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find master nodes security group")
+	}
+	log.Printf("Found master nodes security group %s", masterSgId)
+	workerSgId, err := findSgId(client, "worker", tag)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find worker nodes security group")
+	}
+	log.Printf("Found worker nodes security group %s", workerSgId)
+
+	log.Print("Ensuring pods security group")
+	podSgId, err := ensureSg(client, "kuryr-pods-security-group", tag)
+	log.Printf("Pods security group %s present", podSgId)
+
+	log.Print("Allowing traffic from masters and nodes to pods")
+	err = ensureSgRule(client, podSgId, masterSgId)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to add rule opening traffic from masters")
+	}
+	err = ensureSgRule(client, podSgId, workerSgId)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to add rule opening traffic from workers")
+	}
+	err = ensureSgRule(client, masterSgId, podSgId)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to add rule opening traffic to masters")
+	}
+	err = ensureSgRule(client, workerSgId, podSgId)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to add rule opening traffic to workers")
+	}
+	log.Print("All requried traffic allowed")
+
+	lbClient, err := openstack.NewLoadBalancerV2(provider, gophercloud.EndpointOpts{
+		Region: region,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create Octavia client")
+	}
+
+	// We need first usable IP from services CIDR
+	// This will get us the first one (subnet IP)
+	ip = svcNet.IP
+	// And this second one (first usable)
+	ip[3] |= 0x01
+	ipStr = ip.String()
+	log.Printf("Creating OpenShift API loadbalancer with IP %s", ipStr)
+	lbId, err := ensureLb(lbClient, "kuryr-api-loadbalancer", ipStr, svcSubnetId)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create OpenShift API loadbalancer")
+	}
+	log.Printf("OpenShift API loadbalancer %s present", lbId)
+
+	log.Print("Creating OpenShift API loadbalancer pool")
+	poolId, err := ensureLbPool(lbClient, "kuryr-api-loadbalancer-pool", lbId)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create OpenShift API loadbalancer pool")
+	}
+	log.Printf("OpenShift API loadbalancer pool %s present", poolId)
+
+	log.Print("Creating OpenShift API loadbalancer listener")
+	listenerId, err := ensureLbListener(lbClient, "kuryr-api-loadbalancer-listener", poolId, 443)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create OpenShift API loadbalancer listener")
+	}
+	log.Printf("OpenShift API loadbalancer listener %s present", listenerId)
+
+	// We need to list all master ports and add them to the LB pool
+	log.Print("Creating OpenShift API loadbalancer pool members")
+	r, _ := regexp.Compile("^master-port-[0-9]+$")
+	portList, err := listPortsMatchingPattern(client, tag, r)
```

Comment:
While master nodes are somewhat special now, we do intend for them to be more cattle-like in the future. Does doing this all require the set of master nodes to be constant?

Reply:
If the above comment about moving management of openshift/kubernetes API LB will get resolved, this will get supported automatically by controller. Otherwise I'll need to add tiny logic that will look up for pool members that have no corresponding ports and delete them. If that would be done in reconciliation, we're good.